wordpress-mobile/WordPress-Android
# [Tooling] Reorg paths in promo screenshot lanes

This PR adds lots of small cleaning touches to the `screenshots.rb` lanes, especially the ones related to promo screenshots

### What

 - Remove `download_promo_strings` legacy lane and DRY everything in the more up-to-date `download_metadata_strings` lane
 - Extract out code that was doing some cleanup on the raw screenshot file names, for readability of main lane's code
 - Support separate promo config files for WordPress vs Jetpack, and add them to the `APP_SPECIFIC_VALUES` constant
 - Move the promotional copies from old location in `fastlane/playstoreres` to `fastlane/{jetpack_,}metadata/android/{locale}/…`
 - Move the promo screenshot assets from old location in `fastlane/playstoreres` to `fastlane/screenshots`.
 - Add the `Noto Serif` font to the repo, for easy install. A separate PR to the `release-toolkit` itself will add ability for the `promo_screenshots` action to check if fonts are installed, and prompt the user to install or activate them if missing or deactivated from FontBook.
 - Remove obsolete local legacy `fastlane/actions` and `fastlane/helpers`, now that we're using the ones from `release-toolkit` anyway

### To test

<details><summary>Known Issues, unrelated to this PR but impacting testing the PR</summary>

This is a bit tricky because despite those changes, running `create_promo_screenshots` will still fail and crash in the current state of things, but unrelated to the changes made from this PR… So the goal here will be to test that the changes made here work as expected, and that the crash we might get are only due to the the two following but unrelated things:

 - The config file references the image `1-create-a-site-or-start-a-blog.png` for one of its entry, but the current `WPScreenshotTest` UI tests don't generate such a raw screenshot. So the generation will crash when getting to the step trying to generate that image.
    -  One could remove the two entries in the `fastlane/screenshots/wordpress-config.json` config file (L70–76 and L136–142) before running the test. That should do the trick
    - But I didn't want to do that as part of this PR diff because I'm not sure if the right fix for this in the future will be to remove that entry… or to update the WPScreenshotTest to generate that raw screenshot again, so I'll let testers do that locally before testing
    - Once https://github.com/wordpress-mobile/release-toolkit/pull/429 lands and we start pointing to it, I might consider still keeping the entries about this `1-create-a-site-or-start-a-blog.png` file in the config files… but `# commenting` them (since YAML would now allow us to support that) so that we still have the config entry for it when we finally fix the UI tests to generate that raw screenshot again. Which is also why I kept it around for the time being.
 - There seems to be [a bug in `drawText` with Arabic text, making `drawText` crash when trying to generate the promo screenshots for that locale on an "Out of Bounds" exception](https://github.com/Automattic/drawText/issues/1).
    - One way to work around this in order to test this PR is to only test with a single latin-script locale, like `fr-FR`
</details>

#### Test Steps

 - Ensure you've installed everything needed to run the screenshot lanes, especially `brew install imagemagick` and `bundle install --with screenshots`
 - Run `bundle exec fastlane screenshots app:wordpress locale:fr-FR` to generate raw screenshots for French using both Pixel 3 and Nexus 9. Feel free to omit `locale:` parameter or to use a different comma-separated list of locales if you want to spice things up.
 - Edit the `fastlane/screenshots/wordpress-config.json` file and remove lines L70–76 and L136–142 — i.e. the entries containing the reference to `images/phoneScreenshots/1-create-a-site-or-start-a-blog.png` — to workaround the issue that this raw screenshot is missing (see "known issues" above for details)
 - Run `bundle exec fastlane create_promo_screenshots app:wordpress locale:fr-FR`.
    - You can add `force:true` parameter if you want to skip the confirmation step
    - You can add `skip_download_strings:true` parameter on subsequent calls to avoid re-downloading the latest string translations from GlotPress on every run
  - Check that the final images (2–7) have been generated in `fastlane/metadata/android/fr-FR/images/{phoneScreenshots,tenInchScreenshots}`
